modules/addons/station-auto-name.lua

```diff
@@ -70,12 +70,17 @@ local function station_name_changer(event)
             if item_name then -- prevent errors if something went wrong
                 local item_name2 = item_name:gsub("^%l", string.upper):gsub('-', ' ') -- removing the - and making first letter capital
 
-                local Item_ore_fluid = "item"
-                if item_name == "crude-oil" then
-                    Item_ore_fluid = "fluid"
+                local item_type = 'item'
+                if item_name == 'crude-oil' then
+                    item_type = 'fluid'
                 end
-                --Final string:
-                entity.backer_name = string.format("[L] [img=%s.%s] %s %s (%s)", Item_ore_fluid, item_name, item_name2, entity.backer_name, Angle( entity ))
+
+                entity.backer_name = config.station_name:gsub('__icon__', '[img=' .. item_type .. '.' .. item_name .. ']')
+                    :gsub('__item_name__', item_name2)
+                    :gsub('__backer_name__', entity.backer_name)
+                    :gsub('__direction__', Angle(entity))
+                    :gsub('__x__', math.floor(entity.position.left_top.x))
+                    :gsub('__y__', math.floor(entity.position.left_top.y))
```

Comment:
This code is invalid and raises an error.
```suggestion
                    :gsub('__x__', math.floor(entity.position.x))
                    :gsub('__y__', math.floor(entity.position.y))
```